pandas/core/internals/api.py

```diff
@@ -87,6 +89,14 @@ def make_block(
     - Block.make_block_same_class
     - Block.__init__
     """
+    warnings.warn(
+        # GH#56815
+        "make_block is deprecated and will be removed in a future version. "
+        "Use public APIs instead.",
```

Comment:
Do we want to point to the new `pd.api.internals.create_dataframe_from_blocks`? I know we don't want to encourage that, but TBH I assume this is used by few libraries for a specific reason anyway (exactly the use case for which the pd.api.internals can be used for), and the docstring of that function also asks the user to raise an issue about it